src/dsp_tools/utils/shared.py

```diff
@@ -99,7 +99,7 @@ def try_network_action(
     raise BaseError(failure_msg)
 
 
-def validate_xml_against_schema(input_file: Union[str, etree._ElementTree[Any]]) -> bool:
+def validate_xml_against_schema(input_file: Union[str, Path, etree._ElementTree[Any]]) -> bool:
```

Comment:
again, off topic: I'm not a fan of having these union types - IMO if you find yourself in these situations, it's best to have a method each, and e.g. the `Path` one should read the file, parse it, and then call the `ElementTree` one.  
To re-iterate, a method should to exactly one thing; and when you have these union types, you always do multiple things: find out, which of the types you got; transforming one thing into another; and finally doing the actual job at hand.

Reply:
I understand your concerns, but I'm not a fan of having two slightly different methods... I wish I had good old Java method overloading at hand. I have googled method overloading in Python, and found the following possibilities:
 - https://docs.python.org/3/library/functools.html#functools.singledispatch
 - https://pypi.org/project/overloading/
 - https://pypi.org/project/multipledispatch/
 - https://pypi.org/project/multimethod/
 - https://pypi.org/project/multimethods/
 
 I have tried out the stdlib singledispatch and pushed it. How do you like it?

P.S. It doesn't work yet, because it doesn't support union types in Python 3.9. I would have to upgrade to Python 3.11 to use union types. If you think that singledispatch is generally a good idea, I would upgrade to 3.11.

P.P.S. I would use the union type for `Union[str, Path]`, because these two are very similar. While it's definitely worth writing two methods for str/Path and _ElementTree, it's NOT worth writing two methods for str and Path.

Reply:
I think the `singledispatch` is still better than the initial union implementation, but if there is a downside to tooling, then that's still a downside. And I don't see what the issue is with having two similar methods - at the end of the day, overloading is also just having similar methods in terms of the signature, they just happen to have the same name. So I would still prefer two different methods that don't impact tooling, describe in their name what they really do (`validate_xml_file()` vs. `validate_xml_tree()`), and one happens to call the other.

As for `str` vs. `Path`, I see your point, but the clean way would be to go for one (namely `Path`) and refactor the instances that call the function with `str` to use `Path` instead. Adding complexity to be able not to reduce inconsistency seems like a bad tradeoff, doesn't it? :)

PS: When you have informative method and parameter names, you don't really need docstrings either. When you have a method like this:

```python
validate_xml_file(path: Path) -> bool:
    """
    Validates an XML file against the DSP XML Schema.

    Args:
        path: the path pointing to the XML file

    Returns:
        True if the validation was successful, False otherwise
    """"
    ...
```

The docstrings do not add any value, but make the code harder to read and harder to maintain (having to change two places in code, if functionally one thing changes).